### ISSUE TITLE
Multiple DISCOUNT lines when there are multiple discounts

### ISSUE THREAD

**Issue by joelbrock** [CORE-POS/IS4C#410]
Rather than automatically dropping in a single discount line during the end-of-trans procedures, when there are multiple discounts applied to a trans. insert a DISCOUNT line for each discount applied.

Poss. ripple effects:
-  Reporting queries that use the DISCOUNT line?
-  Counting transactions based on DISCOUNT line?  B/c there is currently only 1 / trans.

**Comment by gohanman** [comment]
84f4194dec96ce2cb239e28dba71b69db694407a

First real attempt at this. Does not deal with the custom situations at this point. I'm just coming up with a total percent discount based on stacking rules.

Setting a discount:
```php
DiscountModule::updateDiscount(new DiscountModule(12, 'custdata'));
```
This adds a 12% discount. The 2nd argument controls whether it adds an additional discount or updates an existing discount.

This will result in a 5% discount:
```php
DiscountModule::updateDiscount(new DiscountModule(12, 'custdata'));
DiscountModule::updateDiscount(new DiscountModule(5, 'custdata'));
```

This will result in a 17% discount (stacking mode) or 12% discount (non-stacking mode)
```php
DiscountModule::updateDiscount(new DiscountModule(12, 'custdata'));
DiscountModule::updateDiscount(new DiscountModule(5, 'NeedBasedDiscount'));
```

At the end of the transaction POS adds log records indicating the discounts that applied with % and $ values:
```
+--------------+-------------------+------------+----------+-------+
| upc          | description       | trans_type | regPrice | total |
+--------------+-------------------+------------+----------+-------+
| DISCLINEITEM | custdata          | L          |      1.2 |    12 | 
| DISCLINEITEM | NeedBasedDiscount | L          |      0.5 |     5 | 
+--------------+-------------------+------------+----------+-------+
```

I want to make sure this works before trying to determine where custom scenarios could tie in.

**Comment by gohanman** [comment]
The stacking/non-stacking system described in the previous post is now in real-life testing.

The beer/wine madness seems to need a separate step. First POS checks all registered discounts in conjunction with stacking rules and decides what the percent discount is. The basic transaction discount is then (discountable items total) * (percent discount).

Then *second*, the registered discounts (or even a separate set of modules) get a chance to adjust the transaction discount *amount* (rather than percentage) based on their own crazypants logic. Making sure this doesn't screw up sales tax collection would be the responsibility of the discount's author.

**Comment by gohanman** [comment]
Multiple percentage discounts was included in 1.2.0.
